SDWebImage/SDImageLoadersManager.m

```diff
@@ -100,4 +100,17 @@ - (BOOL)canRequestImageForURL:(nullable NSURL *)url {
     return nil;
 }
 
+- (BOOL)shouldBlockFailedURLWithURL:(NSURL *)url error:(NSError *)error {
+    NSArray<id<SDImageLoader>> *loaders = self.loaders;
+    for (id<SDImageLoader> loader in loaders.reverseObjectEnumerator) {
+        if (![loader respondsToSelector:@selector(shouldBlockFailedURLWithURL:error:)]) {
```

Comment:
This logic, seems wrong.

We need check `canRequestImageForURL:url` to determine which loader is in used. And then check delegate method.